src/snowflake/cli/_plugins/git/manager.py

```diff
@@ -95,3 +102,33 @@ def _stage_path_part_factory(stage_path: str) -> StagePathParts:
         if stage_path.startswith(USER_STAGE_PREFIX):
             return UserStagePathParts(stage_path)
         return GitStagePathParts(stage_path)
+
+    def _filter_files_list(
+        self, stage_path_parts: GitStagePathParts, files_on_stage: List[str]
+    ) -> List[str]:
+        if not stage_path_parts.directory:
+            return self._filter_supported_files(files_on_stage)
+
+        # Check if file paths starts with git repository name
+        # if not, remove first part of the path
+        if files_on_stage[0].startswith(stage_path_parts.stage_name):
+            stage_path = stage_path_parts.path
+        else:
+            stage_path = stage_path_parts.path_without_repository_name()
```

Comment:
Can we encapsulate this in `stage_path_parts.get_stage_path()`?

Reply:
Needs first file path from stage, `stage_path_parts.get_stage_path(files_on_stage[0])` looks strange. 🫤

Reply:
Outdated